src/services/BookingService.ts

```diff
@@ -0,0 +1,97 @@
+import { Brackets } from "typeorm";
+import bookingRepository from "../repositories/BookingRepository";
+import IRepository from "../types/IRepository";
+import Booking from "../entities/Booking";
+import BookingCreateBodyValidator from "../validators/Booking/BookingCreateBodyValidator";
+import SlotService from "./SlotService";
+import HttpException from "../exceptions/HttpException";
+
+
+class BookingService {
+
+  public getRepository(): IRepository<Booking> {
+    return bookingRepository();
+  }
+
+  public async isSlotAvalible(
+    slotId: number,
+    bookingData: BookingCreateBodyValidator,
+  ): Promise<Booking> {
+
+    const repo = this.getRepository();
+    const slotService = new SlotService();
+
+    const slotRequested = await slotService.getByKey(
+      "id", slotId,
+    );
+
+    let bookingVerificationQuery;
+
+    if (slotRequested.isDisabled === false) {
+      const {
+        startDate, endDate,
+      } = bookingData;
+
+      bookingVerificationQuery = await repo.createQueryBuilder().select("booking")
+        .from(
+          Booking,
+          "booking",
+        )
+        .where(
+          "booking.slotId = :slotId", { slotId },
+        )
+        .andWhere(new Brackets((qb) => {
+          qb.where(
+            "booking.startDate >= :startDate", { startDate },
+          )
+            .orWhere(
+              "booking.endDate >= :endDate", { endDate },
+            );
+        }))
+        .getOne();
+
+      console.log(bookingVerificationQuery);
+    }
+    return bookingVerificationQuery;
+  }
+
+  public async create(
+    slotId: number, bookingData: BookingCreateBodyValidator,
+  ): Promise<Booking> {
+
+
+    const repo = this.getRepository();
+    const slotService = new SlotService();
+
+    const isThereABookingInSlot = await this.isSlotAvalible(
+      slotId, bookingData,
+    );
+
+
+    let booking;
```

Comment:
We need to avoid as much as possible doing this sort of thing (i.e., defining a variable and mutating it based on a conditional) unless we cannot thing of a better way to do so.

In this case, if `isSlotAvailable` method handles all the responsibility of deciding whether to create a booking or not, we can just throw an exception there and no `if` statement is going to be needed here, therefore you can just go ahead with the `return repo.save(...)`.